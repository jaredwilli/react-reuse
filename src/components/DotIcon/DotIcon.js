import React from 'react';
import PropTypes from 'prop-types';

const DotIcon = props => (
    <svg
        xmlns="http://www.w3.org/2000/svg"
        width={props.size}
        height={props.size}
        viewBox="0 0 10 10"
    >
        <circle
            cx="963"
            cy="3070"
            r="5"
            fill={props.fill}
            fillRule="evenodd"
            transform="translate(-958 -3065)"
        />
    </svg>
);

DotIcon.propTypes = {
    /** The size of the dot */
    size: PropTypes.number,
    /** The fill color for the dot */
    fill: PropTypes.string
};

DotIcon.defaultProps = {
    size: 5,
    fill: '#959595'
};

export default DotIcon;
