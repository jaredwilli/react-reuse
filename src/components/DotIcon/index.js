import DotIcon from './DotIcon';

export default DotIcon;
