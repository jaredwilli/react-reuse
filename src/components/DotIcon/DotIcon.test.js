import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import DotIcon from './DotIcon';

describe('DotIcon', () => {
    it('renders the component correctly', () => {
        const tree = renderer.create(<DotIcon />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
