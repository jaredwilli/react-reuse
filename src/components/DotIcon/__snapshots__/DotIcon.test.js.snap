// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotIcon renders the component correctly 1`] = `
<svg
  height={5}
  viewBox="0 0 10 10"
  width={5}
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="963"
    cy="3070"
    fill="#959595"
    fillRule="evenodd"
    r="5"
    transform="translate(-958 -3065)"
  />
</svg>
`;
